association.go

```diff
@@ -519,20 +579,73 @@ func (a *Association) handleInitAck(p *packet, i *chunkInitAck) error {
 	return nil
 }
 
+// The caller should hold the lock.
+func (a *Association) handleHeartbeat(c *chunkHeartbeat) []*packet {
+	a.log.Debug("chunkHeartbeat")
+	hbi, ok := c.params[0].(*paramHeartbeatInfo)
+	if !ok {
+		a.log.Warn("failed to handle Heartbeat, no ParamHeartbeatInfo")
+	}
+
+	return pack(&packet{
+		verificationTag: a.peerVerificationTag,
+		sourcePort:      a.sourcePort,
+		destinationPort: a.destinationPort,
+		chunks: []chunk{&chunkHeartbeatAck{
+			params: []param{
+				&paramHeartbeatInfo{
+					heartbeatInformation: hbi.heartbeatInformation,
+				},
+			},
+		}},
+	})
+}
+
+func (a *Association) handleCookieEcho(c *chunkCookieEcho) []*packet {
+	if !bytes.Equal(a.myCookie.cookie, c.cookie) {
+		return nil
+	}
+
+	// stop T1-init timer
```

Comment:
These comments seem a bit redundant as the code is really clear

Reply:
Fixed at ef7368e